src/viam/sdk/components/board/client.cpp

```diff
@@ -101,7 +102,7 @@ Board::analog_value BoardClient::read_analog(const std::string& analog_reader_na
 
     const grpc::Status status = stub_->ReadAnalogReader(ctx, request, &response);
     if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
+        throw ViamException(status.error_message(), ViamErrorCode::GRPC);
```

Comment:
I'd love to come up with a scheme whereby we could entirely preserve the `grpc::Status` object into the exception.

Reply:
See what I did with the `GRPCException` class.